packages/controllers/src/permissions/rpc-methods/requestPermissions.ts

```diff
@@ -0,0 +1,64 @@
+import type { PermittedHandlerExport } from '@metamask/rpc-methods/types';
+import type {
+  JsonRpcEngineEndCallback,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from 'json-rpc-engine';
+import { nanoid } from 'nanoid';
+import { MethodNames } from '../utils';
+
+import { invalidParams } from '../errors';
+import type { PermissionConstraint, RequestedPermissions } from '../Permission';
+
+export const requestPermissionsHandler: PermittedHandlerExport<
+  RequestPermissionsHooks,
+  [RequestedPermissions],
+  PermissionConstraint[]
+> = {
+  methodNames: [MethodNames.requestPermissions],
+  implementation: requestPermissionsImplementation,
+  methodDescription: 'Request permissions.',
+  hookNames: {
+    requestPermissions: true,
+  },
```

Comment:
I see that it's used on the extension snaps branch. It might be easier to land this without these extra pieces that don't do anything yet. Maybe this can be removed? Or perhaps we can land the rpc-method improvements first seperately?

Reply:
`methodDescription` and `permissionDescription` are removed in a PR to this repo (https://github.com/MetaMask/snaps-skunkworks/pull/130), and `hookNames` and `selectHooks` are implemented in the extension here: https://github.com/MetaMask/metamask-extension/pull/12664